### ISSUE TITLE
Error with restify-router

### ISSUE THREAD

**Issue by henryjw** [DataDog/dd-trace-js#440]
Using
    - dd-trace         0.8.0
    - restify             5.0.1
    - restify-router 0.5.1

It looks like `restify-router` conflicts with the tracer. When I remove `restify-router` initialization, the tracer works as expected. The tracer error causes requests to hang.

Below is a sample of before and after removing the use of the router

```javascript
// Using router (results in error)
const router = new Router();
require('./modules/api-' + key + '/routes').initRoutes(router)
router.applyRoutes(app);
```

```javascript
// Not using router (works fine)
require('./modules/api-' + key + '/routes').initRoutes(app)
```

Stack trace when attempting to make a request.

```
 2019-02-14T16:24:04.983Z v0.39.2 [error] TypeError: fn.apply is not a function
     at Server.<anonymous> (/var/www/node_modules/dd-trace/src/plugins/restify.js:49:15)
     at next (/var/www/node_modules/restify/lib/server.js:997:30)
     at f (/var/www/node_modules/once/once.js:25:25)
     at Server.middleware (/var/www/modules/middleware-forward-erp-call/index.js:24:69)
     at Server.<anonymous> (/var/www/node_modules/dd-trace/src/plugins/restify.js:49:15)
     at next (/var/www/node_modules/restify/lib/server.js:997:30)
     at f (/var/www/node_modules/once/once.js:25:25)
     at Server.middleware (/var/www/modules/middleware-send-data/index.js:26:9)
     at Server.<anonymous> (/var/www/node_modules/dd-trace/src/plugins/restify.js:49:15)
     at next (/var/www/node_modules/restify/lib/server.js:997:30)
     at f (/var/www/node_modules/once/once.js:25:25)
     at Server.middleware (/var/www/modules/middleware-parse-headers/index.js:21:9)
     at Server.<anonymous> (/var/www/node_modules/dd-trace/src/plugins/restify.js:49:15)
     at next (/var/www/node_modules/restify/lib/server.js:997:30)
     at f (/var/www/node_modules/once/once.js:25:25)
     at Server.<anonymous> (/var/www/modules/middleware-jsonapi/index.js:355:9)
     at Server.<anonymous> (/var/www/node_modules/dd-trace/src/plugins/restify.js:49:15)
     at next (/var/www/node_modules/restify/lib/server.js:997:30)
     at f (/var/www/node_modules/once/once.js:25:25)
     at Server.customSend (/var/www/app.js:196:5)
     at Server.<anonymous> (/var/www/node_modules/dd-trace/src/plugins/restify.js:49:15)
     at next (/var/www/node_modules/restify/lib/server.js:997:30)
```

Looks like the root cause is in the `restify` plugin for `dd-trace`. The error surfaces when a route with multiple middlewares is added.

<img width="1506" alt="restify plugin" src="https://user-images.githubusercontent.com/3443648/52802238-8ac08a80-304d-11e9-939b-fd107f602baf.png">

**Comment by rochdev** [comment]
Fixed in 0.9.0

**Comment by henryjw** [comment]
Thanks for the quick turnaround @rochdev. The tracer is working as expected in 0.9.0.
